lib/formats/putty.js

```diff
@@ -52,10 +57,76 @@ function read(buf, options) {
 	if (key.type !== keyType) {
 		throw (new Error('Outer key algorithm mismatch'));
 	}
+
+	si += publicLines;
+	if (lines[si]) {
+		parts = splitHeader(lines[si++]);
+		assert.equal(parts[0].toLowerCase(), 'private-lines');
+		var privateLines = parseInt(parts[1], 10);
+		if (!isFinite(privateLines) || privateLines < 0 ||
+			privateLines > lines.length) {
+			throw (new Error('Invalid private-lines count'));
+		}
+
+		var privateBuf = Buffer.from(
+			lines.slice(si, si + privateLines).join(''), 'base64');
+
+		if (encryption === 'aes256-cbc') {
+			if (!options.passphrase) {
+				throw (new errors.KeyEncryptedError(
+					options.filename, 'PEM'));
+			}
+
+			var iv = Buffer.from(Array
+				.from({length: 16})
+				.map(function () { return 0; }));
+			var decipher = crypto.createDecipheriv(
+				'aes-256-cbc',
+				derivePPK2EncryptionKey(options.passphrase),
+				iv);
+			decipher.setAutoPadding(false);
+			privateBuf = Buffer.concat([
+				decipher.update(privateBuf), decipher.final()]);
+		}
+
+		key = new PrivateKey(key);
+		if (key.type !== keyType) {
+			throw (new Error('Outer key algorithm mismatch'));
+		}
+
+		var sshbuf = new SSHBuffer({buffer: privateBuf});
+		if (alg === 'ssh-dss') {
+			key.addPart('x', sshbuf.readPart());
```

Comment:
See comment on addPart below. I don't like this "construct the private key with only public key fields and add the rest in later" part. Please just do it the same way as the other formats and don't make this one special.

Reply:
Thanks for the review. I'll try to get this sorted in the coming days